src/glb_writer.rs

```diff
@@ -0,0 +1,512 @@
+                    "type" => "SCALAR",
+                    "componentType" => self.accessor_feature_ids.component_type
+                }
+            ],
+            "bufferViews" => array![
+                object!{
+                    "name" => "Positions",
+                    "buffer" => 0,
+                    "byteLength" => self.buffer_view_position.byte_length,
+                    "byteOffset" => self.buffer_view_position.byte_offset,
+                },
+                object!{
+                    "name" => "Colors",
+                    "buffer" => 0,
+                    "byteLength" => self.buffer_view_color.byte_length,
+                    "byteOffset" => self.buffer_view_color.byte_offset,
+                },
+                object!{
+                    "name" => "Feature IDs",
+                    "buffer" => 0,
+                    "byteLength" => self.buffer_view_feature_ids.byte_length,
+                    "byteOffset" => self.buffer_view_feature_ids.byte_offset,
+                },
+                object!{
+                    "name" => "Iteration Count",
+                    "buffer" => 0,
+                    "byteLength" => self.buffer_view_iterations.byte_length,
+                    "byteOffset" => self.buffer_view_iterations.byte_offset,
+                },
+                object!{
+                    "name" => "Point ID",
+                    "buffer" => 0,
+                    "byteLength" => self.buffer_view_point_id.byte_length,
+                    "byteOffset" => self.buffer_view_point_id.byte_offset,
+                },
+                object!{
+                    "name" => "Last xform ID",
+                    "buffer" => 0,
+                    "byteLength" => self.buffer_view_last_xforms.byte_length,
+                    "byteOffset" => self.buffer_view_last_xforms.byte_offset,
+                },
+                object!{
+                    "name" => "Last color xform ID",
+                    "buffer" => 0,
+                    "byteLength" => self.buffer_view_last_color_xforms.byte_length,
+                    "byteOffset" => self.buffer_view_last_color_xforms.byte_offset,
+                }
+            ],
+            "buffers" => array![
+                object!{
+                    "byteLength" => self.buffer_length
+                }
+            ]
+        };
+
+        let json_str = json::stringify(json);
+        let length = json_str.as_bytes().len() as u32;
+        self.json = json_str;
+        self.json_chunk.chunk_length = length;
+        self.json_chunk.padding_length = compute_padding_length(length, ALIGNMENT);
+    }
+
+    fn write_header(&self, file: &mut File) {
+        let error_msg = "could not write glb header";
+        file.write_all(b"glTF").expect(error_msg);
+        file.write_all(&GLTF_VERSION.to_le_bytes()).expect(error_msg);
+        file.write_all(&self.total_length.to_le_bytes()).expect(error_msg);
```

Comment:
ohh looks like I'm forgetting to compute the total length of the GLB file, that's probably one big reason why the GLB is not yet valid.